### PR TITLE
Capital Changes Refactoring

### PR DESCRIPTION
**Pull request by lianga888**
- Allow capital changes to be defined by a target value, in addition to the currently support input of deltas. The delta will be calculated on the portfolio value at the dt of the capital change
- Refactor the checking, calculating and processing of capital changes to better support upstream processes

### CODE REVIEW

File: zipline/finance/performance/period.py [hunk 1]
```diff
@@ -550,8 +548,8 @@ def as_account(self):
             getattr(self, 'day_trades_remaining', float('inf'))
         account.leverage = getattr(self, 'leverage',
                                    period_stats.gross_leverage)
-        account.net_leverage = period_stats.net_leverage
-
+        account.net_leverage = getattr(self, 'net_leverage',
```
Comment (ssanderson) [inline zipline/finance/performance/period.py]: When is this not set?

File: zipline/algorithm.py [hunk 2]
```diff
@@ -785,6 +789,70 @@ def _create_daily_stats(self, perfs):
...
+
+            log.info('Processing capital change to target %s at %s. Capital '
+                     'change delta is %s' % (capital_change['value'], dt,
+                                             capital_change_amount))
+        elif capital_change['type'] == 'delta':
+            capital_change_amount = capital_change['value']
+            log.info('Processing capital change of delta %s at %s'
+                     % (capital_change_amount, dt))
+        else:
+            log.error("Capital change %s does not indicate a valid type "
```
Comment (jdricklefs) [inline zipline/algorithm.py]: What's your thinking on ignoring errors here? This seems pretty serious to me.

Or alternatively, what do you think of moving this validation of the given capital changes earlier in the algorithm's lifecycle so the error condition is recognized before any simulation processing?

File: zipline/finance/performance/period.py (hunk 1 above)
Comment (lianga888) [inline zipline/finance/performance/period.py]: I'm not sure I follow, but previously we always calculated `net_leverage` from the positions zipline records in the position tracker. This change allows us to get a value from the broker, just like the other account attributes

File: zipline/algorithm.py (hunk 2 above)
Comment (lianga888) [inline zipline/algorithm.py]: There doesn't seem to be much validation for similar inputs done in zipline itself. In this case, upstream processes do any necessary validation before the simulation

**Pull request merged**